examples/threading/threading.c

```diff
@@ -0,0 +1,50 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#define MS_TO_US(ms) ((ms) * 1000) // Define the conversion from milliseconds to microseconds
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+
+
+void* threadfunc(void* thread_param)
+{
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    usleep(MS_TO_US(thread_func_args->obtain_wait_time)); 
+    pthread_mutex_lock(thread_func_args->mutex);
```

Comment:
@Rishikesh38 can `pthread_mutex_lock` fail?

Reply:
Yes, it can fail. I should have checked the return value of mutex, if return value == 0 then its working; else it failed